src/repository/repository.cpp

```diff
@@ -0,0 +1,94 @@
+#include "repository/repository.h"
+
+#include "repository/schemamanager.h"
+
+#include "util/assert.h"
+#include "util/logger.h"
+
+
+namespace mixxx {
+
+//static
+const QString Repository::kDefaultDatabaseConnectionName = "MIXXX";
+
+// The schema XML is baked into the binary via Qt resources.
+//static
+const QString Repository::kDefaultSchemaFile(":/schema.xml");
+
+//static
+const int Repository::kRequiredSchemaVersion = 27;
+
+namespace {
+
+const Logger kLogger("Repository");
+
+const QString kDatabaseFileName = "mixxxdb.sqlite";
+
+} // anonymous namespace
+
+Repository::Repository(
```

Comment:
Ok, that's a valid argument. I wanted to be more generic, but it turns out that this class simply host a generic connection (pool) that is initialized with custom parameters related to mixxxdb.sqlite. We might rename it in some final step.